### PR TITLE
[stdlib] Fix conversion from Float16 to unsigned integer types

### PR DESCRIPTION
**Pull request by xwu**
Fixes a think-o, unblocking #36205, which will serve as the test for this change.

Resolves [SR-14293](https://bugs.swift.org/browse/SR-14293).

### CODE REVIEW

**Comment by xwu** [comment]
@swift-ci smoke test

**Comment by xwu** [comment]
cc @benrimmington

File: stdlib/public/core/IntegerTypes.swift.gyb [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ public struct ${Self}
   public init(_ source: ${FloatType}) {
     _precondition(source.isFinite,
       "${FloatType} value cannot be converted to ${Self} because it is either infinite or NaN")
-%     if not (FloatBits == 16 and bits >= 32): # Float16 is always in-range for 32- and 64-bit ints.
+%     if not (FloatBits == 16 and signed and bits >= 32): # Float16 is always in-range for 32- and 64-bit signed integer types.
```
Comment (stephentyrone) [inline stdlib/public/core/IntegerTypes.swift.gyb:1163]: We could tighten this up a bit more and elide the upper-bound check for u16, u32, u64.

**Review by stephentyrone — CHANGES_REQUESTED** [review]

File: stdlib/public/core/IntegerTypes.swift.gyb (hunk 1 above)
Comment (xwu) [reply stdlib/public/core/IntegerTypes.swift.gyb]: Fiiiiiine...

**Comment by xwu** [comment]
@swift-ci smoke test

**Comment by xwu** [comment]
@swift-ci smoke test macOS platform

**Comment by xwu** [comment]
@swift-ci smoke test Windows platform

**Comment by stephentyrone** [comment]
Thanks @xwu!

**Pull request merged**